src/main/java/com/powsybl/openloadflow/network/impl/LfNetworkLoaderImpl.java

```diff
@@ -68,7 +68,7 @@ private static void createVoltageControls(LfNetwork lfNetwork, List<LfBus> lfBus
                 // check target voltage
                 checkUniqueTargetVControllerBus(lfGenerator, controllerTargetV, controllerBus, generatorControlledBus);
 
-                if (lfGenerator.hasVoltageControl()) {
+                if (lfGenerator.hasVoltageControl() && controllerBus.isVoltageControllerEnabled()) {
```

Comment:
This condition can be done before to avoid not usufruit checks ?

Reply:
Done; in fact some refactoring has been done here, to avoid the null/NaN initialization and the useless checks.